### ISSUE TITLE
PyTorch RuntimeError

### ISSUE THREAD

**Issue by zefr0x** [flathub/io.github.voxelcubes.panelcleaner#1]
Output generation failed:

<class 'RuntimeError'>: Traceback (most recent call last):
  File "/app/lib/python3.11/site-packages/pcleaner/gui/worker_thread.py", line 141, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/pcleaner/gui/image_details_driver.py", line 468, in generate_output
    prc.generate_output(
  File "/app/lib/python3.11/site-packages/pcleaner/gui/processing.py", line 175, in generate_output
    ctm.model2annotations_gui(
  File "/app/lib/python3.11/site-packages/pcleaner/gui/ctd_interface_gui.py", line 120, in model2annotations_gui
    process_image(
  File "/app/lib/python3.11/site-packages/pcleaner/ctd_interface.py", line 165, in process_image
    mask, mask_refined, blk_list = model(
                                   ^^^^^^
  File "/app/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/inference.py", line 179, in __call__
    blks = postprocess_yolo(blks, self.conf_thresh, self.nms_thresh, resize_ratio)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/inference.py", line 115, in postprocess_yolo
    det = non_max_suppression(det, conf_thresh, nms_thresh)[0]
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/utils/yolov5_utils.py", line 263, in non_max_suppression
    i = torchvision.ops.nms(boxes, scores, iou_thres)  # NMS
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/torchvision/ops/boxes.py", line 40, in nms
    _assert_has_ops()
  File "/app/lib/python3.11/site-packages/torchvision/extension.py", line 46, in _assert_has_ops
    raise RuntimeError(
RuntimeError: Couldn't load custom C++ ops. This can happen if your PyTorch and torchvision versions are incompatible, or if you had errors while compiling torchvision from source. For further information on the compatible versions, check https://github.com/pytorch/vision#installation for the compatibility matrix. Please check your PyTorch version with torch.__version__ and your torchvision version with torchvision.__version__ and verify if they are compatible, and if not please reinstall torchvision so that it matches your PyTorch install.

---

```python
# Inside flatpak:
>>> torch.version.__version__
'2.1.0+cpu-cxx11-abi'
>>> torchvision.version.__version__
'0.16.0+cpu'
```

They should be compatible.

| torch | torchvision | Python        |
|-------|-------------|---------------|
| 2.1 	| 0.16        | >=3.8, <=3.11 |

> Source: https://github.com/pytorch/vision/tree/315f31527e720999eecbb986679b3177d4ed5e37#installation

**Comment by VoxelCubes** [comment]
Thanks for bringing this up. I'm stumped by this issue, seeing as it worked when I tried it, and the hashes guarantee that the flatpak must've been built the same. I'm updating to a new version right now, maybe it'll sort itself out?

**Comment by VoxelCubes** [comment]
I tried updating both to the latest version and now my test version is getting similar results:
```
flatpak run io.github.voxelcubes.panelcleaner
/app/lib/python3.11/site-packages/transformers/utils/generic.py:441: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
/app/lib/python3.11/site-packages/transformers/utils/generic.py:309: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
Traceback (most recent call last):
  File "/app/bin/pcleaner-gui", line 5, in <module>
    from pcleaner.gui.launcher import launch
  File "/app/lib/python3.11/site-packages/pcleaner/gui/launcher.py", line 16, in <module>
    from pcleaner.gui.mainwindow_driver import MainWindow
  File "/app/lib/python3.11/site-packages/pcleaner/gui/mainwindow_driver.py", line 26, in <module>
    import pcleaner.gui.processing as prc
  File "/app/lib/python3.11/site-packages/pcleaner/gui/processing.py", line 17, in <module>
    import pcleaner.gui.ctd_interface_gui as ctm
  File "/app/lib/python3.11/site-packages/pcleaner/gui/ctd_interface_gui.py", line 11, in <module>
    from pcleaner.comic_text_detector.inference import TextDetector
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/inference.py", line 11, in <module>
    from .basemodel import TextDetBase, TextDetBaseDNN
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/basemodel.py", line 8, in <module>
    from .models.yolov5.yolo import load_yolov5_ckpt
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/models/yolov5/yolo.py", line 3, in <module>
    from ...utils.yolov5_utils import *
  File "/app/lib/python3.11/site-packages/pcleaner/comic_text_detector/utils/yolov5_utils.py", line 10, in <module>
    import torchvision
  File "/app/lib/python3.11/site-packages/torchvision/__init__.py", line 6, in <module>
    from torchvision import _meta_registrations, datasets, io, models, ops, transforms, utils
  File "/app/lib/python3.11/site-packages/torchvision/_meta_registrations.py", line 163, in <module>
    @torch._custom_ops.impl_abstract("torchvision::nms")
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/torch/library.py", line 440, in inner
    handle = entry.abstract_impl.register(func_to_register, source)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/torch/_library/abstract_impl.py", line 30, in register
    if torch._C._dispatch_has_kernel_for_dispatch_key(self.qualname, "Meta"):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: operator torchvision::nms does not exist
```

strange

**Comment by VoxelCubes** [comment]
Going back to the other versions, yeah, it fails now with your error. Amazing.

**Comment by VoxelCubes** [comment]
I think I figured it out, there is an error message right at the start that torch was compiled without libjpeg or so, which probably caused these c++ functions to be missing.
When using the pip link from the torch website, pip uses a slightly different version than doing it manually. That didn't have the issue. Working on updating the flatpak now.

**Comment by VoxelCubes** [comment]
This is now fixed, as soon as version 2.2.0 propagates to the stable repo. You can also try the test version:
```
flatpak install --user https://dl.flathub.org/build-repo/79740/io.github.voxelcubes.panelcleaner.flatpakref
```
Thanks again
